unsafe/src/main/java/org/apache/spark/unsafe/PlatformDependent.java

```diff
@@ -145,21 +147,27 @@ public static void freeMemory(long address) {
 
       long signumOffset = 0;
       long magOffset = 0;
+      long intCompactOffset = 0;
       try {
         signumOffset = _UNSAFE.objectFieldOffset(BigInteger.class.getDeclaredField("signum"));
         magOffset = _UNSAFE.objectFieldOffset(BigInteger.class.getDeclaredField("mag"));
+        intCompactOffset = _UNSAFE.objectFieldOffset(
+          BigDecimal.class.getDeclaredField("intCompact"));
```

Comment:
I think we already reply on some internal API, also ignore the exception that failed to get UNSAFE. We may move to use Decimal128 from Hive in next release.